File: src/main/java/gargoyle/ct/pref/impl/prop/CTPrefFloatProperty.java
package gargoyle.ct.pref.impl.prop;

import gargoyle.ct.pref.CTPreferencesProvider;
import gargoyle.ct.prop.CTNumberProperty;
import org.jetbrains.annotations.NotNull;

public class CTPrefFloatProperty extends CTPrefProperty<Float> implements CTNumberProperty<Float> {
    public CTPrefFloatProperty(@NotNull CTPreferencesProvider provider, String name) {
        this(provider, name, 0);
    }

    public CTPrefFloatProperty(@NotNull CTPreferencesProvider provider, String name, float def) {
        super(Float.class, provider, name, def);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/impl/prop/CTPrefFloatProperty.kt
package gargoyle.ct.pref.impl.prop

import gargoyle.ct.pref.CTPreferencesProvider
import gargoyle.ct.prop.CTNumberProperty

class CTPrefFloatProperty @JvmOverloads constructor(provider: CTPreferencesProvider, name: String, def: Float = 0f) :
    CTPrefProperty<Float>(
        Float::class, provider, name, def
    ), CTNumberProperty<Float>
